### PR TITLE
Sitemap Rebuild Workflow

### PR DESCRIPTION
**Pull request by echen-adobe**
Describe your specific features or fixes

Implements a github actions workflow for rebuilding the staging sitemap when a PR is made. This action should help detect issues related to sitemap configs in the future.

Resolves:  https://jira.corp.adobe.com/projects/MWPW/issues/MWPW-157407

Testing Instructions:

To test this locally, 
1) create a .secrets file in the root of the project directory. The secrets file will contain vault role IDs and vault secret IDs, it does NOT contain the API key
2) Ask me for the .secrets content
3) Install `act `locally
4) Run `act -j run-sitemap-script --secret-file .secrets` in your terminal

The terminal should tell you sitemaps were built if it succeeded.

Test URLs:
- Before: https://main--express--adobecom.hlx.page/express/
- After: https://sitemap-action--express--adobecom.hlx.page/express/

### CODE REVIEW

File: test/helpers/sitemap.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    try {
+      const parsedURL = url.split('https://www.adobe.com/')[1];
+      const response = await fetch(baseURL + parsedURL, {
+        method: 'POST',
+        headers: {
+          authorization: `token ${apiKey}`,
+        },
+      });
+      if (response.status !== 200) {
+        console.error(`Error: ${url}`, response.statusText);
```
Comment (fullcolorcoder) [inline test/helpers/sitemap.js:26]: Can you /* eslint-disable these console.log() lines

**Review by fullcolorcoder — CHANGES_REQUESTED** [review]
- Is there a ticket for this 
- What are your testing instructions?

File: test/helpers/sitemap.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  'https://www.adobe.com/express/sitemap.xml',
+];
+const apiKey = process.env.HELIX_API_KEY;
+if (!apiKey) {
+  throw new Error(`Invalid API Key: ${apiKey}`);
+}
+async function makeRequests() {
+  const baseURL = 'https://admin.hlx.page/sitemap/adobecom/express/stage/';
+  const failedURLs = [];
+  for await (const url of urls) {
```
Comment (JingleH) [inline test/helpers/sitemap.js:16]: once we have added the other urls, maybe do promise.all/allSettled to have them run in parallel and save us some Action time?

File: test/helpers/sitemap.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    console.error('Failed to build some urls');
+    for (const url of failedURLs) {
+      console.error(url);
+    }
+    console.error('------------');
+    throw new Error('Failed to rebuild some urls');
+  }
+}
+
+makeRequests();
```
Comment (JingleH) [inline test/helpers/sitemap.js:54]: sometimes this way can generate some UnhandledPromiseRejection warning. Using main().catch(console.error()) might be safer

**Review by JingleH — CHANGES_REQUESTED** [review]

File: test/helpers/sitemap.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+dotenv.config();
+const urls = [
+  'https://www.adobe.com/express/sitemap.xml',
+];
+const apiKey = process.env.HELIX_API_KEY;
+if (!apiKey) {
+  throw new Error(`Invalid API Key: ${apiKey}`);
+}
+async function makeRequests() {
+  const baseURL = 'https://admin.hlx.page/sitemap/adobecom/express/stage/';
```
[the remark this answers is not in the document]
Comment (echen-adobe) [reply test/helpers/sitemap.js]: I think it should support it in theory, but in practice the main sitemaps are always the ones updated.

**Review by fullcolorcoder — APPROVED** [review]

**Review by JingleH — APPROVED** [review]
also uploaded the secrets for now

**Pull request merged**